### ISSUE TITLE
false positive: no-member when importing from matplotlib.cm

### ISSUE THREAD

**Issue by Teagum** [pylint-dev/pylint#9425]
### Bug description

```python
import matplotlib as mpl

x = mpl.cm.viridis(1)
```

### Configuration

```ini
[MAIN]
extension-pkg-allow-list=matplotlib.cm
unsafe-load-any-extension=yes

[MESSAGES CONTROL]
disable=missing-docstring
enable=no-member


[TYPECHECK]
generated-members=matplotlib.cm.viridis
```

### Command used

```shell
pylint a.py
```

### Pylint output

```python
************* Module a
a.py:3:4: E1101: Module 'matplotlib.cm' has no 'viridis' member (no-member)
```

### Expected behavior

No error

### Pylint version

```shell
pylint 3.0.3
astroid 3.0.2
Python 3.12.0 (main, Nov 10 2023, 13:07:45) [GCC 12.2.0]
```

### Additional dependencies

```python
matplotlib==3.8.2
```

**Comment by Pierre-Sassoulas** [comment]
Duplicate of #2289 that was closed as won't fix. If someone want to work on this, it's a matplotlib brain, probably in an external pylint-plugin.

> This is not something that pylint supports out of the box, and usually we require an astroid brain plugin in order to patch these values for the offending module. There's also another choice, which is to use --generated-members as in pylint ... --generated-members=coolwarm.

https://github.com/pylint-dev/pylint/issues/2289#issuecomment-405165838
